server/types.py

```diff
@@ -1,5 +1,24 @@
-from typing import NamedTuple
+from typing import NamedTuple, Optional
 
-# Represents a peer IP address
-Address = NamedTuple('Address', [('host', str), ('port', int)])
-Address.from_string = lambda s: Address(*(s.split(':')))
+
+class Address(NamedTuple):
+    """A peer IP address"""
+
+    host: str
+    port: int
+
+    @classmethod
+    def from_string(cls, s: str) -> "Address":
+        host, port = s.split(":")
+        return cls(host, int(port))
+
+
+# TODO: Find an appropriate place for this
```

Comment:
As long as there is a `server.types` module, I think this is a good place to put it.